### DISCUSSION TITLE
Welcome to opengeode Discussions!

### DISCUSSION THREAD

**Discussion by maxime-esa** [esa/opengeode#80] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gugomea** [comment]
Is there a way to do type casting in opengeode?

**Comment by maxime-esa** [comment]
You can cast a real to an integer with the `fix` operator and the reverse with the `float` operator.
But you can't cast e.g. an OCTET STRING into a SEQUENCE.

https://taste.tuxfamily.org/wiki/index.php?title=Technical_topic:_OpenGEODE_-_SDL_Operators:_How_to_work_with_data#INTEGER_types

**Reply by gugomea** [reply]
If I define two integer types with different ranges, can I cast between them?

**Reply by maxime-esa** [reply]
You don't need to. There will only be a warning.

**Reply by gugomea** [reply]
I remember having trouble with signed vs unsiged values. Is there a way to do some workaround for this situations?

**Reply by maxime-esa** [reply]
Indeed, this is because signed and unsigned types don't mix very well at code level. You can't use the `abs` operator to transform a signed into unsigned because the corresponding operator at code level returns the same type as the parameter. 
There is a workaround however that seems to work: use the `power` operator. 
e.g. if `a` is unsigned and `b` is signed, both expressions will work:
``` 
  a := power(b, 1)
  b := power(a, 1)
```
